lib/eb/server.js

```diff
@@ -206,15 +232,18 @@ module.exports = function () {
                 return callback();
         });
 
-        client.on('disconnect', (data, callback) => {
+        client.on('disconnect', (reason) => {
+            
+            console.log(`[worker.disconnect] ${client.id} has disconnected. Reason: '${reason}', Nodes: ${workerNodes.length}`);
+
             workerNodes.some((worker, index) => {
```

Comment:
to be discussed if this is the right moment to remove the worker from the workerNodes array and emit an error on the jobs on that node in the queue

Reply:
solution would be to regularly check for workers (e.g. in 'statistics') and only remove and fail jobs if the worker did not re-connect within n minutes.